### PR TITLE
Fix TimeDistanceMatrix results sequence

### PR DESCRIPTION
**Pull request by DmitryAk**
# Issue

TimeDistanceMatrix results sequence in vector was broken for the case when `sources count` > `targets count`. E.g., if sources == {A,B,C} and targets == {D,E}, it is expected to return a vector
```
{A->D A->E
 B->D B->E
 C->D C->E}
```
but actually before this fix the result was:
```
{A->D B->D
 C->D A->E
 B->E C->E}
```

This PR adds a unit test reproducing the issue + fixes the issue.

## Tasklist

 - [x] Add tests
 - [x] Add #fixes with the issue number that this PR addresses
 - [x] Update the [changelog](CHANGELOG.md)

### CODE REVIEW

File: test/gurka/test_time_tracking.cc [hunk 1]
```diff
@@ -254,6 +255,27 @@ TEST(TimeTracking, routes) {
...
+  d.Parse(matrix.c_str());
+  rapidjson::Value a = d["sources_to_targets"].GetArray()[0].GetArray();
+
+  // check the timings
+  times.clear();
+  for (rapidjson::SizeType i = 0; i < a.Size(); ++i) {
+    times.push_back(a[i]["time"].GetDouble());
+  }
+
+  const std::vector<double> expectedTimes = {51, 193}; // {"A"->"D", "A"->"E"}
```
Comment (DmitryAk) [inline test/gurka/test_time_tracking.cc:277]: before the fix: actual values were `{51, 34}`

**Comment by nilsnolde** [comment]
I’ll take a look today at this one, been around the matrix code recently anyways.

File: test/gurka/test_time_tracking.cc [hunk 2]
```diff
@@ -254,6 +255,27 @@ TEST(TimeTracking, routes) {
...
+      boost::format(
+          R"({"costing":"auto","sources":[{"lon":%1%,"lat":%2%},{"lon":%3%,"lat":%4%},{"lon":%5%,"lat":%6%}],"targets":[{"lon":%7%,"lat":%8%},{"lon":%9%,"lat":%10%}]})") %
+      map.nodes["A"].first % map.nodes["A"].second % map.nodes["B"].first % map.nodes["B"].second %
+      map.nodes["C"].first % map.nodes["C"].second % map.nodes["D"].first % map.nodes["D"].second %
+      map.nodes["E"].first % map.nodes["E"].second;
+  std::string matrix = actor.matrix(req.str(), nullptr, &api);
+  rapidjson::Document d;
+  d.Parse(matrix.c_str());
+  rapidjson::Value a = d["sources_to_targets"].GetArray()[0].GetArray();
+
```
Comment (nilsnolde) [inline test/gurka/test_time_tracking.cc:270]: This is a bit of weird place for testing matrix stuff. I'd suggest to glue it to the end of `test/matrix.cc` where the others live too. It's not gurka map enabled but there's other requests already.

**Review by nilsnolde — COMMENTED** [review]
Thanks a lot @DmitryAk , in fact that bug must've been there forever! More sources than targets is a more unusual case, yet a fully valid one. I'd just think about putting the test to the other matrix tests. Other than that it's good for me.

File: test/gurka/test_time_tracking.cc (hunk 2 above)
Comment (DmitryAk) [reply test/gurka/test_time_tracking.cc:270]: I'll try to move the test into `test/matrix.cc` today, thank you.

Comment (DmitryAk) [reply test/gurka/test_time_tracking.cc]: Done, I've just added a new test into `test/matrix.cc`, and removed the changes from `gurka/test_time_tracking.cc`.

**Review by kevinkreiser — APPROVED** [review]

**Pull request merged**